src/virtual_elements.js

```diff
@@ -87,30 +87,38 @@ const elementOpen = function(tag, key, statics, var_args) {
    */
   const attrsArr = data.attrsArr;
   const newAttrs = data.newAttrs;
-  let attrsChanged = false;
+  const isNew = !attrsArr.length;
```

Comment:
This makes me thnk, what if we don't create the Array in NodeData, but instead, if there is no array, treat it as new and preallocate it with the correct size (`new Array(arguments.length - ATTRIBUTES_OFFSET`)? I think that this should help avoid re-allocations, especially if the element has quite a few attributes.

I think we could move the creation of `newAttrs` and `attrs` here as well, to avoid creating them for `Text` nodes.